### PR TITLE
Remove .NET Core 3.1 support and standardize on .NET 6

### PR DESCRIPTION
**Pull request by yash-nisar**
# Description

Remove .NET Core 3.1 support and standardize on .NET 6

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: https://github.com/dapr/dotnet-sdk/issues/1036

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

**Comment by cgillum** [comment]
@halspang how can we move forward with this? Microsoft has already announced that .NET Core 3.1 is no longer supported.

https://devblogs.microsoft.com/dotnet/net-core-3-1-will-reach-end-of-support-on-december-13-2022/

In addition to Microsoft's communication, does there need to be a proposal in a Dapr community forum? I'd like to move this forward, if possible, to reduce support burden and test matrix.

File: test/Dapr.E2E.Test/DaprTestApp.cs [hunk 1]
```diff
@@ -128,11 +128,7 @@ private static string GetTargetFrameworkName()
         {
             var targetFrameworkName = ((TargetFrameworkAttribute)Assembly.GetExecutingAssembly().GetCustomAttributes(typeof(TargetFrameworkAttribute), false).FirstOrDefault()).FrameworkName;
             string frameworkMoniker;
-            if (targetFrameworkName == ".NETCoreApp,Version=v3.1")
-            {
-                frameworkMoniker = "netcoreapp3.1";
-            }
-            else if (targetFrameworkName == ".NETCoreApp,Version=v5.0")
+            if (targetFrameworkName == ".NETCoreApp,Version=v5.0")
```
Comment (halspang) [inline test/Dapr.E2E.Test/DaprTestApp.cs:131]: Isn't NET 5 also gone?

File: .github/workflows/itests.yml [hunk 2]
```diff
@@ -19,14 +19,8 @@ jobs:
...
+        dotnet-version: ['6.0', '7.0']
         include:
-        - dotnet-version: '3.1'
-          install-3: true
-          display-name: '.NET Core 3.1'
-          framework: 'netcoreapp3.1'
-          prefix: 'netcoreapp31'
-          install-version: '3.1.x' # We always need a new .NET
         - dotnet-version: '6.0'
           install-3: false
```
Comment (halspang) [inline .github/workflows/itests.yml:25]: We won't need the `install-3` anymore, will we?

File: .github/workflows/sdk_build.yml [hunk 3]
```diff
@@ -42,14 +42,8 @@ jobs:
...
+        dotnet-version: ['6.0', '7.0']
         include:
-        - dotnet-version: '3.1'
-          install-3: true
-          display-name: '.NET Core 3.1'
-          framework: 'netcoreapp3.1'
-          prefix: 'netcoreapp31'
-          install-version: '3.1.x' # We always need a new .NET
         - dotnet-version: '6.0'
           install-3: false
```
Comment (halspang) [inline .github/workflows/sdk_build.yml:48]: Same here.

**Review by halspang — CHANGES_REQUESTED** [review]

File: test/Dapr.E2E.Test/DaprTestApp.cs (hunk 1 above)
Comment (yash-nisar) [reply test/Dapr.E2E.Test/DaprTestApp.cs:131]: Yup, fixed.

File: .github/workflows/itests.yml (hunk 2 above)
Comment (yash-nisar) [reply .github/workflows/itests.yml:25]: Nope, fixed.

File: .github/workflows/sdk_build.yml (hunk 3 above)
Comment (yash-nisar) [reply .github/workflows/sdk_build.yml:48]: Fixed.

**Review by halspang — APPROVED** [review]

**Pull request merged**